browser/src/app/TilesMiddleware.ts

```diff
@@ -1417,6 +1417,45 @@ class TileManager {
 		return this.tiles[key];
 	}
 
+	public static overlapInvalidatedRectangleWithView(
+		part: number,
+		mode: number,
+		wireId: number,
+		invalidatedRectangle: cool.SimpleRectangle,
+		textMsg: string,
+	) {
+		let needsNewTiles = false;
+
+		for (const key in TileManager.tiles) {
+			const coords: TileCoordData = TileManager.get(key).coords;
+			const tileRectangle = [coords.x, coords.y, this.tileSize, this.tileSize];
+
+			if (
```

Comment:
B) but in the changed version it is:

if (
        coords.part === part && 
        coords.mode === mode && 
        invalidatedRectangle.intersectsRectangle(tileRectangle)
) {
        if (app.isRectangleVisibleInTheDisplayedArea(tileRectangle)) {
                TileManager.invalidateTile(key, wireId);
                needsNewTiles = true;   
        }
}

and TileManager.invalidateTile(key, wireId); has ended up inside the inner condition. I feel the invalidateTile should be outside the inner condition.